code/modules/ninja/suit/ninjaDrainAct.dm

```diff
@@ -107,37 +107,48 @@
 
 	return drain_total
 
-//RDCONSOLE//
-/obj/machinery/computer/rdconsole/ninjadrain_act(obj/item/clothing/suit/space/space_ninja/ninja_suit, mob/living/carbon/human/ninja, obj/item/clothing/gloves/space_ninja/ninja_gloves)
+//RD SERVER//
+/obj/machinery/rnd/server/master/ninjadrain_act(obj/item/clothing/suit/space/space_ninja/ninja_suit, mob/living/carbon/human/ninja, obj/item/clothing/gloves/space_ninja/ninja_gloves)
 	if(!ninja_suit || !ninja || !ninja_gloves)
 		return INVALID_DRAIN
 
 	. = DRAIN_RD_HACK_FAILED
 
-	to_chat(ninja, span_notice("Hacking \the [src]..."))
-	AI_notify_hack()
+	// If the traitor theft objective is still present, this will destroy it...
+	if(source_code_hdd)
+		to_chat(ninja, span_notice("Hacking \the [src]..."))
+		AI_notify_hack()
+		to_chat(ninja, span_notice("Encrypted source code detected. Overloading storage device..."))
+		if(do_after(ninja, 30 SECONDS, target = src) && src)
```

Comment:
```suggestion
		if(do_after(ninja, 30 SECONDS, target = src) && !QDELETED(src))
```

`QDELETED` will check null as well, but will catch more cases.

Reply:
Though also, don't you not need this? I just checked and `do_after` checks this same thing.

Reply:
Copy-pasta from somewhere else in code. Removing it.